### ISSUE TITLE
Imported target "torch" includes non-existent path "C:/Program Files/NVIDIA Corporation/NvToolsExt/include"

### ISSUE THREAD

**Issue by deeponcology** [pytorch/pytorch#21572]
## 🐛 Bug

Hello, 
It seems there is a hardcoded path inside CmakeLists relating to **NvToolsExt**. 
In CUDA 10.0 there is NO NvToolsExt folder while Pytorch is expecting one. 
Specifically, when using CUDA 10.1 with the latest version of TorchLib on Windows 10, I get the following exception. 

> Make Error in CMakeLists.txt:
  Imported target "torch" includes non-existent path
    "C:/Program Files/NVIDIA Corporation/NvToolsExt/include"

Indeed, there is no Include directory under that folder. 
![image](https://user-images.githubusercontent.com/36111600/59157321-74818300-8ab1-11e9-8e24-7b88a7af1c2c.png)

## To Reproduce

1. Download https://github.com/tobiascz/MNIST_Pytorch_python_and_capi
2. Download the CUDA supported version 10.0 of TorchLib for windows 10 
3. Link against TorchLib from within Cmake 
4. The output from CMake:

`"C:\Program Files\JetBrains\CLion 2019.1.4\bin\cmake\win\bin\cmake.exe" -DCMAKE_BUILD_TYPE=Release -G "CodeBlocks - NMake Makefiles" F:\Dropbox\Tech\python\startups\deeponcology-new\production\CPP\mnist
-- The C compiler identification is MSVC 19.16.27031.1
-- The CXX compiler identification is MSVC 19.16.27031.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE  
-- Found CUDA: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0 (found version "10.0") 
-- Caffe2: CUDA detected: 10.0
-- Caffe2: CUDA nvcc is: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0/bin/nvcc.exe
-- Caffe2: CUDA toolkit directory: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0
-- Caffe2: Header version is: 10.0
-- Found CUDNN: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0/include  
-- Found cuDNN: v7.6.0  (include: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0/include, library: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0/lib/x64/cudnn.lib)
-- Autodetected CUDA architecture(s):  6.1
-- Added CUDA NVCC flags for: -gencode;arch=compute_61,code=sm_61
-- Found torch: F:/libtorch/lib/torch.lib  `

## Expected behavior
There should be no hardcoded path inside CmakeLists.

## Environment

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux): Windows 10
 - How you installed PyTorch (`conda`, `pip`, source): TorchLib 
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version: 10.0 or 10.1 
 - GPU models and configuration:
 - Any other relevant information:

## Additional context

See also https://github.com/pytorch/pytorch/issues/11447

**Comment by peterjc123** [comment]
Yes, we do hardcode the PATH for the NVTX library. However, we check the environment variable `NVTOOLSEXT_PATH` first, and the hardcoded one is only used as a fallback value. See https://github.com/pytorch/pytorch/blob/master/caffe2/CMakeLists.txt#L633-L644(there is a bug in this one. I'll fix this one later.)  and https://github.com/pytorch/pytorch/blob/master/tools/setup_helpers/nvtoolext.py#L17-L23. Would you please check if that variable exists in your environment? If not, then it's likely you have an incomplete installation of CUDA. Have you installed VS before installing CUDA?

**Comment by deeponcology** [comment]
Indeed, a faulty CUDA installation.  Thanks.

**Comment by mojiayuan** [comment]
> 确实，错误的CUDA安装。谢谢。

How did you solve it

**Comment by peterjc123** [comment]
@mojiayuan Ensure you have a correct CUDA installation and make sure you have set the environment variable `NVTOOLSEXT_PATH` correctly.

**Comment by mojiayuan** [comment]
> @mojiayuan Ensure you have a correct CUDA installation and make sure you have set the environment variable `NVTOOLSEXT_PATH` correctly.

I have installed CUDA correctly, but I still lack the include folder under the folder "C:\Program Files\NVIDIA Corporation\NvToolsExt". Then I copied the include folder under the libtorch to the folder "C:\Program Files\NVIDIA Corporation\NvToolsExt". Cmake compiled successfully, but ran error in VS2015.

**Comment by peterjc123** [comment]
You can get the NVTX installer in a folder named "VisualStudioExtension" after extracting the cuda installer using something like 7zip.

**Comment by mojiayuan** [comment]
> 使用7zip之类的解压缩cuda安装程序后，可以在名为“ VisualStudioExtension”的文件夹中获得NVTX安装程序。

Thanks, I've solved it

**Comment by mojiayuan** [comment]
The model of Pytorch training is transformed into the model of libtorch call with lower accuracy，Why is that.?


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "pytorch/pytorch"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年8月11日(星期二) 上午9:14
收件人:&nbsp;"pytorch/pytorch"<pytorch@noreply.github.com&gt;;
抄送:&nbsp;"START"<844417548@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [pytorch/pytorch] Imported target "torch" includes non-existent path "C:/Program Files/NVIDIA Corporation/NvToolsExt/include" (#21572)





 
You can get the NVTX installer in a folder named "VisualStudioExtension" after extracting the cuda installer using something like 7zip.

**Comment by peterjc123** [comment]
@mojiayuan Please open a new issue.
